packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -55,7 +57,14 @@ export class PrometheusExporter implements MetricExporter {
    */
   constructor(config: ExporterConfig = {}, callback?: () => void) {
     this._logger = config.logger || new api.NoopLogger();
-    this._port = config.port || PrometheusExporter.DEFAULT_OPTIONS.port;
+    this._host =
+      config.host ||
+      process.env.OTEL_EXPORTER_PROMETHEUS_HOST ||
```

Comment:
why not using the environment from core for that ? We have there all environments and defaults.

Reply:
Original PR took this route but please see https://github.com/open-telemetry/opentelemetry-js/pull/1857#discussion_r561681920 where it was suggested that it be moved into the package itself.

I'm open to both options, please advise.

Reply:
I think it doesn't scale well to have env names (and their defaults) in core for all plugins/exporters/resource detectors/... out somewhere.

Reply:
I think there is a middle ground we can take. The important part of @obecny's suggestion is to use `getEnv` from core which provides functionality for both web and node. In general this is a good idea, but doesn't really matter for this exporter which can't be used in web anyways. @Flarna is also right that it doesn't scale to have env names in core. We should keep getEnv in the core, but move the names to the packages which use them.

Reply:
> I think it doesn't scale well to have env names (and their defaults) in core for all plugins/exporters/resource detectors/... out somewhere.

no1 said to have it for all but for those that are already in core and are treated as a "core" packages.

Reply:
IMHO 3rd party exporters/... should be move to a different repo like it was done with several plugins already.

Reply:
prometheus, zipkin, and jaeger are required by spec. imo the main repo should be able to be a spec compliant sdk on its own without contrib.

Reply:
what's the final decision on this then? 

If I understand correctly, we want to leverage getEnv(), but its weird to have 3rd party env defined in the core package, however jaeger, zipkin and prometheus are treated special as they are required by spec and so may make sense to have them there finally? On the other hand this has 3 approvals and is ready to go functionally.

Reply:
@naseemkullah tbh I don't know , I'm confused as well by the responses what are the expectations :/ @Flarna @dyladan  ?

Reply:
I still think that each plugin/exporter should encapsulate it's internals as good as possible.
If reading environment variables is complicated because of platform dependencies than using shared code is a good choice for using a shard component to encapsulate the reading.
In general I think default values for configurations (e.g. ports/hosts/...) are internals. This changes if they are part of the spec. In that case it's better to move the defaults into something generated like semantic constants.

But if others prefer to have the defaults outside of of the plugin I'm also fine with it.

Reply:
Alright thanks everyone for the feedback! Keeping it as-is for this first iteration, happy to revisit where the env vars live anytime.

Reply:
Since this package is node-only I think it is fine to access environment vars though `process.env`. The `getEnv` helper exists primarily to be able to share configs with node and web which in this case isn't important

Reply:
In general i agree with @Flarna the configs and their associated defaults for a package should live in that package.